Sources/Tools/PsiStudio/Microsoft.Psi.PsiStudio/PsiStudioSettings.cs

```diff
@@ -101,6 +102,11 @@ public PsiStudioSettings()
         /// </summary>
         public bool ShowTimingRelativeToSelectionStart { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to set any open dataset object into autosave mode.
+        /// </summary>
+        public bool AutoSaveDataset { get; set; }
```

Comment:
I would suggest using the plural AutoSaveDatasets here, it feels more general and also perhaps in the future we might allow opening multiple datasets